File: IrrgartenRuby/irrgarten/fuzzy_player.rb
#encoding utf-8

require_relative 'dice'

module Irrgarten
    class FuzzyPlayer < Player
        #Como al final creo uno a partir de un jugador resucitado, 
        #he decidido implementar la copia en el constructor
        def initialize (other)
            copia(other)
        end

        def move(direction, valid_moves)
            direction_preference = super(direction, valid_moves)
            return Dice.next_step(direction_preference, valid_moves, get_intelligence)
        end

        def attack
            return Dice.intensity(@strength + sum_weapons)
        end

        def to_s
            return "Fuzzy " + super
        end

        protected
        def defensive_energy
          return Dice.intensity(@intelligence + sum_shields)
        end
    end
end

File: IrrgartenJava/src/irrgarten/FuzzyPlayer.java
package irrgarten;
import java.util.ArrayList;

/**
 *
 * @author Juan Carlos Vílchez Sánchez
 */

public class FuzzyPlayer extends Player {
    public FuzzyPlayer(Player other){
        super(other);
    }
    
    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        Directions directionReturned = super.move(direction, validMoves);
        return Dice.nextStep(directionReturned, validMoves, getIntelligence());
    }
    
    @Override
    public float attack(){
        return Dice.intensity(getStrength()) + this.sumWeapons();
    }
    
    @Override
    protected float defensiveEnergy(){
        return Dice.intensity(getIntelligence()) + this.sumShields();
    }
    
    @Override
    public String toString(){
        return "Fuzzy " + super.toString();
    }
}
